autobot-slm-backend/tests/test_cleanup_never_destroys_data_14856.py

```
@@ -898,6 +939,152 @@ def test_wrong_node_undefined_fact_would_hard_error_rather_than_delete() -> None
     assert removed is False, "an undefined role fact reached the delete branch — this is #14856 itself"
 
 
+# --------------------------------------------------------------------------
+# #15822: the token the gate reads must survive set_fact, whatever its spelling
+# --------------------------------------------------------------------------
+def _leaves(node: Any, path: tuple = ()):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `_leaves`

(ANN202)